src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        Random randGen = new Random();
+        int figurePos = randGen.nextInt(FiguresSet.values().length);
+        FiguresSet figureElem = FiguresSet.values()[figurePos];
+        ColorSupplier colorSupplier = new ColorSupplier();
+        String color = colorSupplier.getRandomColor();
+        Figure randomFigure = null;
+        switch (figureElem) {
+            case SQUARE:
+                int side = randGen.nextInt(20) + 1;
+                randomFigure = new Square(color, side);
+                break;
+            case RECTANGLE:
+                int sideA = randGen.nextInt(20) + 1;
+                int sideB = randGen.nextInt(20) + 1;
+                randomFigure = new Rectangle(color, sideA, sideB);
+                break;

```

Comment:
```suggestion
                return new Rectangle(color, sideA, sideB);
```
можна одразу повертати значення з методу